scripts/zones/Horlais_Peak/mobs/Fighting_Sheep.lua

```diff
@@ -12,6 +12,21 @@ entity.onMobInitialize = function(mob)
     mob:setMod(xi.mod.ICE_MEVA, 75) -- Todo: Move to mob_resists.sql
 end
 
+entity.onMobSpawn = function(mob)
+    mob:SetMobSkillAttack(701)
+end
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    mob:setLocalVar("skill_tp", mob:getTP())
+end
+
+entity.onMobWeaponSkill = function(target, mob, skill)
+    if skill:getID() == 274 then
+        mob:addTP(mob:getLocalVar("skill_tp"))
+        mob:setLocalVar("skill_tp", 0)
+    end
+end
```

Comment:
I don't understand whats happening here? Is this to preserve the mob's TP while using the special melee mobskills?

If the mob's TP is being erased on these special melee attacks, that's a bug that we need to fix, rather than for you to work around

Reply:
https://github.com/LandSandBoat/server/issues/1198

Reply:
if the previous conversation had not been resolved, you might have seen where I already mentioned this but with more detail. the fact github keeps allowing people other than us to resolve the conversations is maddening.

Reply:
https://github.com/LandSandBoat/server/pull/2378#discussion_r928151545

Reply:
If you wanted to get your hands dirty, I think the fix here is something like this:
https://github.com/LandSandBoat/server/blob/base/src/map/ai/controllers/mob_controller.cpp#L349-L352

```cpp

    if (currentDistance <= PMobSkill->getDistance())
    {
        auto mobSkill = MobSkill(PActionTarget->targid, PMobSkill->getID());
        if (wsList == PMob->getMobMod(MOBMOD_ATTACK_SKILL_LIST))
        {
            mobSkill.setTP(0);
        }
        return mobSkill;
    }
```

Reply:
@dallano 
I want you to try 
```sql
INSERT INTO `mob_skills` VALUES (274,18,'sheep_charge',0,7.0,2000,1000,4,16,0,0,0,0,0); -- Sheep charge as melee attack
```
Adding that to its skill list for its melee attack, and tell me if it still strips TP. If it does, then try 20 where the 16 is, and try again. Normally for these, we would zero out that 1000 to silence the message (Iron Giants etc) but per the video, the message is included so this should retain it.

Reply:
This does prevent TP loss, yes. However they're not gaining any TP from attacking. Which I guess would be expected. Another side effect is it has caused the damage to go from ~40 to ~400

Reply:
> This does prevent TP loss, yes. However they're not gaining any TP from attacking. Which I guess would be expected. Another side effect is it has caused the damage to go from ~40 to ~400

we may need to rename this 2nd instance (I'd go with `'sheep_charge_melee'`) and adjust its script a bit. I also suspect the tp gain from attacking with "special" melee like this may need handled on a case by case basis anyway - some might have different gains than usual or lack thereof, but at the script level we can just `mob:addTP()` as needed if we go that route. Other LSB members may have differing opinions and I am partly thinking out loud ..er.. in text. There hasn't been a major effort to study the tp gains on different mobs with this yet. I wouldn't worry about that yet. The dmg drop is more of a concern as nothing here should have changed that at all.